runtime/codert_vm/cnathelp.cpp

```diff
@@ -3240,12 +3240,16 @@ old_slow_jitNewInstanceImplAccessCheck(J9VMThread *currentThread)
 			 * unless the two classes are in the same nest (JDK11 and beyond).
 			 */
 #if JAVA_SPEC_VERSION >= 11
-			if (NULL == thisClass->nestHost) {
+			if ((areExtensionsEnabled(currentThread->javaVM) && J9ROMCLASS_IS_OPTIONNESTMATE_SET(thisClass->romClass))
+				|| (NULL == thisClass->nestHost)
```

Comment:
In each altered condition, why do we not do the simpler test (`NULL == thisClass->nestHost`) first?

Reply:
We can also do the check `(NULL == thisClass->nestHost)` first. If you prefer that way, I can change the code to do `(NULL == thisClass->nestHost)` first.

Reply:
Changed the order of the checks.